chain/chain/src/chain.rs

```diff
@@ -1246,15 +1248,12 @@ impl Chain {
     /// We validate the header but we do not store it or update header head
     /// based on this. We will update these once we get the block back after
     /// requesting it.
-    pub fn process_block_header(
-        &self,
-        header: &BlockHeader,
-        on_challenge: &mut dyn FnMut(ChallengeBody),
-    ) -> Result<(), Error> {
+    pub fn process_block_header(&self, header: &BlockHeader) -> Result<(), Error> {
         debug!(target: "chain", "Process block header: {} at {}", header.hash(), header.height());
 
         check_known(self, header.hash())?.map_err(|e| Error::BlockKnown(e))?;
-        self.validate_header(header, &Provenance::NONE, on_challenge)?;
+        let mut challenges = vec![];
+        self.validate_header(header, &Provenance::NONE, &mut challenges)?;
```

Comment:
Hm, so ignoring challenges here seems surprising. I think that:

* this is correct refactor (the code was ignoring challenges before)
* this probably is a correct behavior (the only challenge we get here is double sign, and comment there explicitly says that that's OK). 

still, looks surprising. Let's maybe add  comment saying that yes, we ignore challenges delibrately here, and do want to proceed even if there are challenges. 

I also wonder if it makes sense to change `validate_header` to  

```
fn validate_header() -> Result<Option<DoubleSign>, ()>
```

to enforce in types that the only possible challenge here is double sign. If that refactor makes sense (I don't know :) ), could you file an issue?

Reply:
+1 for adding the comment that explains why we ignore them for now (you could also add the TODO with the challenges issue number - 2445)

Reply:
I'll add comments and file an issue. This function is only used in ClientActor::process_block. I think the logic there can be simplified greatly after we make process_block async. I'll add an issue.

Reply:
https://github.com/near/nearcore/issues/6965